pkg/autoscaler/autoscaler/pdplan_autoscaler.go

```diff
@@ -130,35 +131,108 @@ func (am *autoScalerManager) updateAutoscalingClusters(tac *v1alpha1.TidbCluster
 }
 
 func (am *autoScalerManager) createAutoscalingClusters(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler, groupsToCreate []string, groupPlanMap map[string]pdapi.Plan) error {
-	// TODO in next PR
-	return nil
-}
+	var errs []error
+	for _, group := range groupsToCreate {
+		plan := groupPlanMap[group]
+		component := plan.Component
 
-func (am *autoScalerManager) updateTidbMonitor(tm *v1alpha1.TidbMonitor) error {
-	ns := tm.GetNamespace()
-	tmName := tm.GetName()
-	monitorSpec := tm.Spec.DeepCopy()
-
-	// don't wait due to limited number of clients, but backoff after the default number of steps
-	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		var updateErr error
-		_, updateErr = am.cli.PingcapV1alpha1().TidbMonitors(ns).Update(tm)
-		if updateErr == nil {
-			klog.Infof("TidbMonitor: [%s/%s] updated successfully", ns, tmName)
-			return nil
+		if !checkAutoscalingComponent(tac, component) {
+			continue
 		}
-		klog.V(4).Infof("failed to update TidbMonitor: [%s/%s], error: %v", ns, tmName, updateErr)
-		if updated, err := am.tmLister.TidbMonitors(ns).Get(tmName); err == nil {
-			// make a copy so we don't mutate the shared cache
-			tm = updated.DeepCopy()
-			tm.Spec = *monitorSpec
-		} else {
-			utilruntime.HandleError(fmt.Errorf("error getting updated TidbMonitor %s/%s from lister: %v", ns, tmName, err))
+
+		resource, ok := tac.Spec.Resources[plan.ResourceType]
+		if !ok {
+			errs = append(errs, fmt.Errorf("unknown resource type %v for group %s tac[%s/%s]", plan.ResourceType, plan.Labels[groupLabelKey], tac.Namespace, tac.Name))
+			continue
+		}
+		resList := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.CPU,
+			corev1.ResourceMemory: resource.Memory,
+		}
+
+		autoTcName, err := genAutoClusterName(tac, component, plan.Labels, resource)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		autoTc := &v1alpha1.TidbCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      autoTcName,
+				Namespace: tc.Namespace,
+				OwnerReferences: []metav1.OwnerReference{
+					controller.GetTiDBClusterAutoscalerOwnerRef(tac),
+				},
+			},
+			Status: v1alpha1.TidbClusterStatus{
+				AutoScaler: &v1alpha1.TidbClusterAutoScalerRef{
+					Name:      tac.Name,
+					Namespace: tac.Namespace,
+				},
+			},
+			Spec: v1alpha1.TidbClusterSpec{
+				Cluster: &v1alpha1.TidbClusterRef{
+					Name:      tc.Name,
+					Namespace: tc.Namespace,
+				},
+			},
+		}
+
+		switch component {
+		case v1alpha1.TiKVMemberType.String():
+			resList[corev1.ResourceStorage] = resource.Storage
+
+			autoTc.Spec.TiKV = tc.Spec.TiKV.DeepCopy()
+			autoTc.Spec.TiKV.Replicas = int32(plan.Count)
+			autoTc.Spec.TiKV.ResourceRequirements = corev1.ResourceRequirements{
+				Limits:   resList,
+				Requests: resList,
+			}
+			if autoTc.Spec.TiKV.Config == nil {
+				autoTc.Spec.TiKV.Config = &v1alpha1.TiKVConfig{
```

Comment:
We have to do the following:

- DeepCopy the `tc.spec` of the original TC, reset the `spec.pd/tidb/tiflash/pump/ticdc`
- Update replicas, resources and labels of `spec.tikv`

Reply:
Similar logic for TiDB